ucsb-cs156-w24/proj-courses-w24-5pm-4
# BUG: Deleting a personal schedule doesn't delete associated PSCourses

Changed files:
* src/main/java/edu/ucsb/cs156/courses/controllers/PersonalScheduleController.js
* src/test/java/edu/ucsb/cs156/courses/controllers/PersonalScheduleControllerTests.js

This pull request addresses the bug that won’t delete the associated courses when you delete a personal schedule.
The issue was caused by how in the PersonalScheduleController.js, in the controller delete a personal schedule, it did not iterate through the PSCourse repository to delete the associated courses. So, to fix the bug, within the function I added code to iterate through the repository to find the associated courses that was connected to the Personal Schedule by ID, and deleted the courses that had the same ID.

Steps to test:
- Create your own personal schedule, add multiple courses and sections to the schedule
- Delete the personal schedule, check if the associated courses are still in the ‘All Courses’ Tab

Personal Schedule created with courses:
<img width="1440" alt="Screenshot 2024-03-03 at 6 59 14 PM" src="https://github.com/ucsb-cs156-w24/proj-courses-w24-5pm-4/assets/129819259/f670528e-099b-4ca8-b322-eb2a5da4a02d">

Personal Schedule Deleted:
<img width="1440" alt="Screenshot 2024-03-03 at 6 59 34 PM" src="https://github.com/ucsb-cs156-w24/proj-courses-w24-5pm-4/assets/129819259/2a49e26b-0d3f-47a5-91c1-1375f982b875">


No Courses left in All Courses Tab:
<img width="1440" alt="Screenshot 2024-03-03 at 6 59 42 PM" src="https://github.com/ucsb-cs156-w24/proj-courses-w24-5pm-4/assets/129819259/e3cb25de-9682-4e7d-9531-89dee5436c0d">



Test initial bug at: https://courses.dokku-00.cs.ucsb.edu/
Dokku site at: https://proj-courses-megumi-ondo-dev.dokku-08.cs.ucsb.edu

Closes #8
